accepted/future-releases/unified-collections/feature-specification.md

```diff
@@ -0,0 +1,871 @@
+
+    ```dart
+    main() {
+      [await for (_ in stream)]; // Error.
+    }
+    ```
+
+*   `await` is used before a C-style `for` element. `await` can only be used
+    with `for-in` loops.
+
+    ```dart
+    main() {
+      [await for (;;)]; // Error.
+    }
+    ```
+
+*   The type of the condition expression (the second clause) in a C-style `for`
+    element may not be assigned to `bool`.
+
+    ```dart
+    [for (; "not bool";) 1] // Error.
+    ```
+
+## Constant Semantics
+
+The runtime semantics below are also used to determine the compile-time value of
+a collection literal marked `const`, with a few restrictions:
+
+*   A `listLiteral` is constant if it occurs in a constant context or if it is
+    directly prefixed by `const`. If so, then its `elements` must all be
+    constant elements.
+
+*   A `setOrMapLiteral` is constant if it occurs in a constant context or if it
+    is directly prefixed by `const`. If so, then its `elements` must all be
+    constant elements.
+
+*   An `expressionElement` is a constant element if its expression is a constant
+    expression, and a potentially constant element if it's expression is a
+    potentially constant expression.
+
+*   A `mapEntry` is a constant element if both its key and value expression are
+    constant expressions, and a potentially constant element if both are
+    potentially constant expressions.
+
+*   A `spreadElement` starting with `...` is a constant element if its
+    expression is constant and it evaluates to a constant `List`, `Set` or `Map`
+    instance originally created by a list, set or map literal. It is a
+    potentially constant element if the expression is potentially constant
+    expression.
+
+*   A `spreadElement` starting with `...?` is a constant element if its
+    expression is constant and it evaluates to `null` or a constant `List`,
+    `Set` or `Map` instance originally created by a list, set or map literal. It
+    is a potentially constant element if the expression is potentially constant
+    expression.
+
+*   An `ifElement` is constant if its condition is a constant expression
+    evaluating to a Boolean value and either:
+
+    *   If the condition evaluates to `true`, the "then" element is a constant
+        expression, and any "else" element is a potentially constant element.
+
+    *   If the condition evaluates to `false`, then the "then" element is a
+        potentially constant element and any "else" element is a constant
+        element.
+
+*   An `ifElement` is potentially constant if its condition, "then" element, and
+    "else" element (if any) are potentially constant expressions.
+
+*   A `forElement` is never a constant element. A `for` element cannot be used
+    in constant collection literals.
+
+*   It is a compile-time error if any element in a constant set or key in a
+    constant map does not have a primitive operator `==`.
+
+*   It is a compile-time error if an element in a const set is equal to any
```

Comment:
That sentence doesn't actually make sense :)
To be an element in a const set, the const set must already exist. The set cannot contain two elements that are equal, so this can never happen. 
You must check for the equality on values *before* putting them into a set.

So, I think this should be covered where you build the set, when you have the values independently of them being in a set (yet). I'll write that there.

Reply:
Done.